hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedTableMetadata.java

```diff
@@ -101,10 +102,15 @@ public class TestHoodieBackedTableMetadata extends TestHoodieMetadataBase {
   private static final Logger LOG = LoggerFactory.getLogger(TestHoodieBackedTableMetadata.class);
 
   @ParameterizedTest
-  @ValueSource(booleans = {true, false})
-  public void testTableOperations(boolean reuseReaders) throws Exception {
+  @CsvSource({"true,6", "true,8", "false,6", "false,8"})
+  public void testTableOperations(boolean reuseReaders, int tableVersion) throws Exception {
     HoodieTableType tableType = HoodieTableType.COPY_ON_WRITE;
-    init(tableType);
+    initPath();
+    HoodieWriteConfig config = getWriteConfigBuilder(HoodieFailedWritesCleaningPolicy.EAGER, true, true, false, true, false)
+        .build();
+    config.setValue(HoodieTableConfig.VERSION, String.valueOf(tableVersion));
+    config.setValue(HoodieWriteConfig.WRITE_TABLE_VERSION, String.valueOf(tableVersion));
```

Comment:
we only need to set the `write table version`. can you avoid setting the `table version` directly.

Reply:
at the end of every test, can we also add validation of table version in table config in meta client.

Reply:
Addressed. There were few modifications required for it in HoodieTestUtils.